firmware/shared/device/mcp/mcp4461.hpp

```diff
@@ -144,30 +146,30 @@ class MCP4461 : public DigitalPotentiometerController<uint8_t> {
     }
 
     void WriteRegister(Register reg, uint8_t value) {
-        // MCP4461 uses 8-bit command + data format
-        uint8_t cmd[] = {static_cast<uint8_t>(static_cast<uint8_t>(reg) << 4)};
-        uint8_t data[] = {value};
+        // MCP4461 uses 8-bit command + 8-bit data format in a single write
+        uint8_t reg_write[] = {static_cast<uint8_t>(static_cast<uint8_t>(reg) << 4), value};
 
-        i2c::Message msg1 =
-            i2c::Message(device_addr_, cmd, i2c::MessageType::Write);
-        i2c::Message msg2 =
-            i2c::Message(device_addr_, data, i2c::MessageType::Write);
-        i2c_.Write(msg1);
-        i2c_.Write(msg2);
+        i2c::Message msg =
+            i2c::Message(device_addr_, reg_write, i2c::MessageType::Write);
+        
+        i2c_.Write(msg);
     }
 
     uint8_t ReadRegister(Register reg) {
-        uint8_t cmd[] = {static_cast<uint8_t>(static_cast<uint8_t>(reg) << 4)};
+        uint8_t cmd[] = {static_cast<uint8_t>((static_cast<uint8_t>(reg) << 4) | POT_READ)};
         uint8_t data[2] = {};
 
         i2c::Message msg1 =
             i2c::Message(device_addr_, cmd, i2c::MessageType::Write);
         i2c::Message msg2 =
             i2c::Message(device_addr_, data, i2c::MessageType::Read);
+        
         i2c_.Write(msg1);
         i2c_.Read(msg2);
 
-        return data[0];
+        
+        //data is returned as 2 bytes, but for 8-bit resoultion the msb (data[0]) is empty
```

Comment:
Damn that must have been an annoying bug
```suggestion
        // Data is returned as 2 bytes, but for 8-bit resolution the MSB, data[0], is empty.
```